src/main/scala/services/parser/TimeTableParser.scala

```diff
@@ -0,0 +1,54 @@
+package services.parser
+
+import java.util.regex.Pattern
+
+import entities.School
+import entities.module.{Module, RequiredSession}
+import services.Utils
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+
+object TimeTableParser {
+  val TimeTablePattern = Pattern.compile("\\\"?[0-9]L[\\s]([\\S\\s]*?)\\([\\S\\s]+?\\)\\/[0-9]+\\\"?,([A-Za-z]+),([0-9]{1,2}:[0-9]{1,2}),([0-9]{1,2}:[0-9]{1,2}),([\\S\\s]*?),[\\S\\s]*?,([\\S\\s]*?),([\\S\\s]*?),")
+  val SchoolPattern = Pattern.compile("[A-Z][0-9]{4},([\\S\\s]*?),")
+
+  val timeTableCsvStr = Source.fromFile(getClass.getResource("/input/Gregory_Mitten_Full_Timetable.csv").getPath).mkString
+
+  val schools: Map[String, School] = {
+    val schoolMatcher = SchoolPattern.matcher(timeTableCsvStr)
+    var schools = mutable.Set[School]()
+    var schoolId = 0
+
+    while (schoolMatcher.find) {
+      schools += new School(schoolId, Utils.toSnake(schoolMatcher.group(1)), null)
+      schoolId += 1
+    }
+
+    schools.groupBy(_.schoolName).map(_._2.head).filter(!_.schoolName.contains("(")).map(s => s.schoolName -> s).toMap
```

Comment:
Current approach leaves many spaces in the id's.
If the id's were assigned after the data was filtered, then they will be sequential.
e.g.
var schools = mutable.Set[String]()
while (schoolMatcher.find) {
  schools += Utils.toSnake(schoolMatcher.group(1))
}
schools.filter(!_.contains("(")).zipWithIndex.map(s => s._1 -> new School(s._2, s._1, null)).toMap